src/pybind/mgr/dashboard/controllers/nvmeof.py

```diff
@@ -352,13 +353,18 @@ def create(
             self,
             nqn: str,
             rbd_image_name: str,
+            size: str,
             rbd_pool: str = "rbd",
             create_image: Optional[bool] = True,
-            size: Optional[int] = 1024,
             block_size: int = 512,
             load_balancing_group: Optional[int] = None,
             gw_group: Optional[str] = None,
         ):
+            try:
+                size = convert_to_bytes(size, MB)
+            except ValueError as e:
+                raise DashboardException("invalid value for 'size'", code=400) from e
```

Comment:
this could break the UI and the clients as well as you mentioned an it'd make it their responsibility to adapt it. But the problem is we usually sent size at its smallest unit to the APIs which means that if a user select a for 1GB, we convert that to bytes and sent it to API. If API is doing another conversion that might break things. Ideally the API should be backward compatible so it wouldn't cause any behavior changes.

Can this exist in a way that it only needs to be done for the CLI? or maybe an extra argument (or within the current `size`) that takes the unit and converts based on it?

Reply:
That makes sense. I changed it to work with bytes by default - which is an acceptable behavior for the CLI as well. So it won't break the UI :)

Reply:
sounds good, thank you